streamer/src/tls_certificates.rs

```diff
@@ -1,58 +1,98 @@
-    const ED25519_IDENTIFIER: [u32; 4] = [1, 3, 101, 112];
-    let mut private_key = Vec::<u8>::with_capacity(34);
-    private_key.extend_from_slice(&[0x04, 0x20]); // ASN.1 OCTET STRING
-    private_key.extend_from_slice(keypair.secret().as_bytes());
-    let key_pkcs8 = pkcs8::PrivateKeyInfo {
-        algorithm: AlgorithmIdentifier {
-            oid: ObjectIdentifier::from_arcs(&ED25519_IDENTIFIER).expect("Failed to convert OID"),
-            parameters: None,
-        },
-        private_key: &private_key,
-        public_key: None,
-    };
-    let key_pkcs8_der = key_pkcs8
-        .to_der()
-        .expect("Failed to convert keypair to DER")
-        .to_der();
+    // https://www.rfc-editor.org/rfc/rfc8410#section-7
+    //
+    // The hardcoded prefix decodes to the following ASN.1 structure:
+    //
+    //   PrivateKeyInfo SEQUENCE (3 elem)
+    //     version Version INTEGER 0
+    //     privateKeyAlgorithm AlgorithmIdentifier SEQUENCE (1 elem)
+    //       algorithm OBJECT IDENTIFIER 1.3.101.112 curveEd25519 (EdDSA 25519 signature algorithm)
+    //     privateKey PrivateKey OCTET STRING (34 byte)
+    const PKCS8_PREFIX: [u8; 16] = [
+        0x30, 0x2e, 0x02, 0x01, 0x00, 0x30, 0x05, 0x06, 0x03, 0x2b, 0x65, 0x70, 0x04, 0x22, 0x04,
+        0x20,
+    ];
+    let mut key_pkcs8_der = Vec::<u8>::with_capacity(PKCS8_PREFIX.len() + 32);
+    key_pkcs8_der.extend_from_slice(&PKCS8_PREFIX);
+    key_pkcs8_der.extend_from_slice(keypair.secret().as_bytes());
 
-    let rcgen_keypair = rcgen::KeyPair::from_der(&key_pkcs8_der)?;
+    // Create a dummy certificate. Only the SubjectPublicKeyInfo field
+    // is relevant to the peer-to-peer protocols. The signature of the
+    // X.509 certificate is deliberately invalid. (Peer authenticity is
+    // checked in the TLS 1.3 CertificateVerify)
 
-    let mut cert_params = CertificateParams::default();
-    cert_params.subject_alt_names = vec![SanType::IpAddress(san)];
-    cert_params.alg = &rcgen::PKCS_ED25519;
-    cert_params.key_pair = Some(rcgen_keypair);
-    cert_params.distinguished_name = DistinguishedName::new();
-    cert_params
-        .distinguished_name
-        .push(DnType::CommonName, "Solana node");
+    let mut cert_der = Vec::<u8>::with_capacity(0xf4);
+    //    Certificate SEQUENCE (3 elem)
+    //      tbsCertificate TBSCertificate SEQUENCE (8 elem)
+    //        version [0] (1 elem)
+    //          INTEGER  2
+    //        serialNumber CertificateSerialNumber INTEGER (62 bit)
+    //        signature AlgorithmIdentifier SEQUENCE (1 elem)
+    //          algorithm OBJECT IDENTIFIER 1.3.101.112 curveEd25519 (EdDSA 25519 signature algorithm)
+    //        issuer Name SEQUENCE (1 elem)
+    //          RelativeDistinguishedName SET (1 elem)
+    //            AttributeTypeAndValue SEQUENCE (2 elem)
+    //              type AttributeType OBJECT IDENTIFIER 2.5.4.3 commonName (X.520 DN component)
+    //              value AttributeValue [?] UTF8String Solana
```

Comment:
Can we keep using the old "Solana node" -- it is more appropriate. It is issued by this node -- not Solana networks.